### PR TITLE
fix importing string values from log parameters (no ticket)

### PR DESCRIPTION
**Pull request by geishm-ansto**
**Description of work.**
Fixed an issue importing string values from the hdf file that are listed in the parameters file for the instrument.

**To test:**

The unit tested has been updated to included string values - previously only needed numerical values.

The fix does not affect the documentation or needs to be documented.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by geishm-ansto** [comment]
@martyngigg Hi, I still do not have the ability to kick-off a rebuild and it appears your restart didn't go through either as the details on the failure are still dated May-5 which was the original build. Confused?

File: Framework/DataHandling/src/LoadBBY.cpp [hunk 1]
```diff
@@ -498,21 +515,40 @@ void LoadBBY::loadInstrumentParameters(
...
+          // check the default for value numeric and string
+          auto updateOk = false;
+          if (!hdfTag.empty()) {
+            if (isNumeric(details[2])) {
+              if (loadNXDataSet(entry, hdfTag, tmpFloat)) {
+                auto factor = std::stod(details[1]);
+                logParams[logTag] = factor * tmpFloat;
+                updateOk = true;
+              }
+            } else {
```
Comment (gemmaguest) [inline Framework/DataHandling/src/LoadBBY.cpp:535]: Could use `else if` here to reduce the nesting slightly

File: Framework/DataHandling/src/LoadBBY.cpp [hunk 2]
```diff
@@ -498,21 +515,40 @@ void LoadBBY::loadInstrumentParameters(
...
             if (!hdfTag.empty())
-              g_log.warning() << "Cannot find hdf parameter "
-
-                              << hdfTag << ", using default.\n";
+              g_log.warning() << "Cannot find hdf parameter " << hdfTag
+                              << ", using default.\n";
           }
-
-        } catch (const std::invalid_argument &) {
+        } catch (std::invalid_argument &) {
```
Comment (gemmaguest) [inline Framework/DataHandling/src/LoadBBY.cpp:551]: Why was the `const` removed? There seems to be a mixture of styles in this file but it's usually better to pass by const reference unless there's a reason not to.

**Review by gemmaguest — COMMENTED** [review]
I'm not sure of the context here so can't comment on whether it works, really, other than that it does indeed load string parameters. The code looks fine - just a couple of minor tidying suggestions.

File: Framework/DataHandling/src/LoadBBY.cpp (hunk 2 above)
Comment (geishm-ansto) [reply Framework/DataHandling/src/LoadBBY.cpp]: My slip up - restored the const.

File: Framework/DataHandling/src/LoadBBY.cpp (hunk 1 above)
Comment (geishm-ansto) [reply Framework/DataHandling/src/LoadBBY.cpp]: Dropped one level by using else if {}. Thanks.

**Review by gemmaguest — APPROVED** [review]
Thanks for the changes. Looks good to me :+1:

**Pull request merged**